src/Microsoft.Health.Fhir.Liquid.Converter/Hl7v2/Hl7v2Processor.cs

```diff
@@ -48,6 +49,8 @@ public string Convert(string data, string rootTemplate, ITemplateProvider templa
             var hl7v2Data = _dataParser.Parse(data);
             var context = CreateContext(templateProvider, hl7v2Data);
             var rawResult = RenderTemplates(template, context);
+            rawResult = Hl7v2EscapeSequenceProcessor.Unescape(rawResult, hl7v2Data.EncodingCharacters);
```

Comment:
Can we do this in parsing instead of post-processing?

Reply:
The special character escaping is used to represent semantical characters, if we parse them while parsing, they will be used to cut segments(or components) instead of being showed as text.

Reply:
By offline discussion, I'll add this part in the parser.